assignment-4.md

```diff
@@ -1,19 +1,24 @@
 # Assignment 4: Application State with Redux
 
 Up to now you have structured your application in a way that holds the state of
-the individual components locally. You learned how you can use Redux to create a
-global state that handles every state aspect of your application. You now have
-the task to create these building blocks:
+the individual components locally. We will now prepare for a global state
+handling solution powered by *Redux*.
 
-* actions
-* action creators
-* reducer(s) (if you want you can create more than one reducer, but this is up
-  to you)
+We'll start out with the state handling itself, *without integrating it into the
+React app yet*.
 
-Since reducers are always pure functions without side effects it is very
-efficient and easy to unit test them. Write some unit tests that make sure they
-work as intended!
+Create the actions and a reducer for everything that can happen in the app:
 
-The point of this assignment is to *only* create the state management logic
-(*actions*, *action creators*, *reducers*) and its unit tests. This
-assignment is not yet about using this actual data in your application.
+* temperature changes
+* humidity changes
+* reset of minimum/maximum values
+
+Even though the *Redux* code is not yet an active part of our app, we can still
+verify that it works by writing unit tests for it. Remember that *Redux* is
+based on pure functions, which is very convenient in tests: the output is always
+fully defined by the input.
+
+Focus on tests for the *reducer*, which takes in some state and an action, and
+produces a new state as result from that. Think of various scenarios that can
+occur in the app and write them down as test code, asserting the expected new
+state after some action happens with a given starting stating.
```

Comment:
I would rewrite the last line as follows:

Think of various scenarios that can occur in the app and document them as test code. Starting with a mock initial state, assert the new state matches what is expected after an action has been reduced.

It's much more Reduxy ;-)

Reply:
Alright, sounds good, but I'll slightly adapt.

Need to drop "mock", as it's not a mock! It's a real state (but not real estate ;-)).

Reply:
badumm tss!

Reply:
![image](https://user-images.githubusercontent.com/1330022/100014014-0c724a00-2dd6-11eb-9597-6061bb67518f.png)